numba/tests/test_unicode.py

```diff
@@ -2701,5 +2710,347 @@ def impl5():
+        self.assertEqual(base_cfunc('4f5aff66', 19), 4294967296)
+        self.assertEqual(base_cfunc('3723ai4g', 20), 4294967296)
+        self.assertEqual(base_cfunc('281d55i4', 21), 4294967296)
+        self.assertEqual(base_cfunc('1fj8b184', 22), 4294967296)
+        self.assertEqual(base_cfunc('1606k7ic', 23), 4294967296)
+        self.assertEqual(base_cfunc('mb994ag', 24), 4294967296)
+        self.assertEqual(base_cfunc('hek2mgl', 25), 4294967296)
+        self.assertEqual(base_cfunc('dnchbnm', 26), 4294967296)
+        self.assertEqual(base_cfunc('b28jpdm', 27), 4294967296)
+        self.assertEqual(base_cfunc('8pfgih4', 28), 4294967296)
+        self.assertEqual(base_cfunc('76beigg', 29), 4294967296)
+        self.assertEqual(base_cfunc('5qmcpqg', 30), 4294967296)
+        self.assertEqual(base_cfunc('4q0jto4', 31), 4294967296)
+        self.assertEqual(base_cfunc('4000000', 32), 4294967296)
+        self.assertEqual(base_cfunc('3aokq94', 33), 4294967296)
+        self.assertEqual(base_cfunc('2qhxjli', 34), 4294967296)
+        self.assertEqual(base_cfunc('2br45qb', 35), 4294967296)
+        self.assertEqual(base_cfunc('1z141z4', 36), 4294967296)
+
+        # tests with base 0
+        # this fails on 3.0, but in 2.x the old octal syntax is allowed
+        self.assertEqual(base_cfunc(' 0o123  ', 0), 83)
+        self.assertEqual(base_cfunc(' 0o123  ', 0), 83)
+        self.assertEqual(base_cfunc('000', 0), 0)
+        self.assertEqual(base_cfunc('0o123', 0), 83)
+        self.assertEqual(base_cfunc('0x123', 0), 291)
+        self.assertEqual(base_cfunc('0b100', 0), 4)
+        self.assertEqual(base_cfunc(' 0O123   ', 0), 83)
+        self.assertEqual(base_cfunc(' 0X123  ', 0), 291)
+        self.assertEqual(base_cfunc(' 0B100 ', 0), 4)
+
+        # without base still base 10
+        self.assertEqual(cfunc('0123'), 123)
+        self.assertEqual(base_cfunc('0123', 10), 123)
+
+        # tests with prefix and base != 0
+        self.assertEqual(base_cfunc('0x123', 16), 291)
+        self.assertEqual(base_cfunc('0o123', 8), 83)
+        self.assertEqual(base_cfunc('0b100', 2), 4)
+        self.assertEqual(base_cfunc('0X123', 16), 291)
+        self.assertEqual(base_cfunc('0O123', 8), 83)
+        self.assertEqual(base_cfunc('0B100', 2), 4)
+
+    def test_basic_raise_error(self):
+        self.disable_leak_check()
+
+        cfunc = njit(int_usecase)
+        base_cfunc = njit(int_base_usecase)
+
+        # Test conversion from strings and various anomalies
+        for s, v in self.L_might_raise_error:
+            for sign in "", "+", "-":
+                for prefix in "", " ", "\t", "  \t\t  ":
+                    ss = prefix + sign + s
+                    vv = v
+                    if sign == "-" and v is not ValueError:
+                        vv = -v
+                    try:
+                        self.assertEqual(cfunc(ss), vv)
+                    except ValueError:
+                        pass
```

Comment:
For the cases `cfunc(ss)` raises an exception, can you check the exception message to see if it is what you're expecting.

Reply:
This one is tricky, I will try to revise it.